scripts/release/release.sh

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env bash
+
+# Exit script as soon as a command fails.
+set -o errexit
+
+log() {
+  # Print to stderr to prevent this from being 'returned'
+  echo "$@" > /dev/stderr
+}
+
+current_version() {
+  echo "v$(node --print --eval "require('./package.json').version")"
+}
+
+current_release_branch() {
+  v="$(current_version)"
+  echo "release-${v%%-rc.*}"
+}
+
+assert_current_branch() {
+  current_branch="$(git symbolic-ref --short HEAD)"
+  expected_branch="$1"
+  if [[ "$current_branch" != "$expected_branch" ]]; then
+    log "Current branch '$current_branch' is not '$expected_branch'"
+    exit 1
+  fi
+}
+
+push_release_branch_and_tag() {
+  git push upstream "$(current_release_branch)" "$(current_version)"
+}
+
+push_and_publish() {
+  dist_tag="$1"
+
+  log "Pushing release branch and tags to upstream"
+  push_release_branch_and_tag
+
+  log "Publishing package on npm"
+  npm publish --tag "$dist_tag" --otp "$(prompt_otp)"
+
+  if [[ "$dist_tag" == "latest" ]]; then
+    npm dist-tag rm --otp $(prompt_otp) openzeppelin-solidity next
+  fi
+}
+
+prompt_otp() {
+  log -n "Enter npm 2FA token: "
+  read otp
+  echo $otp
+}
+
+environment_check() {
+  if [[ "upstream" != *"$(git remote)"* ]]; then
```

Comment:
Have you tested this line? It doesn't seem to work as you intended. Here's an alternative:

```suggestion
  if ! git remote get-url upstream &> /dev/null; then
```

Reply:
I did test it, it checks if there is a remote named `upstream`. Yours is clearer though.

Reply:
I tested it and it didn't work for me. Might have been a coincidence that it worked in your test environment.